### DISCUSSION TITLE
No plugins detected when scanning VST3 folder

### DISCUSSION THREAD

**Discussion by pieterdd** [monocasual/giada#511] [Q&A]
I downloaded Giada 0.16.2 and I tried to scan for VSTs. I let it scan `/Library/Audio/Plug-Ins/VST3` and no VSTs were found. This is strange because I checked `ls /Library/Audio/Plug-Ins/VST3`. I have plenty of VSTs there, including Vital which should be able to serve as a VSTi.

Any clues on why it's not detecting anything? I'm on macOS 11.6 Big Sur.

**Comment by gvnnz** [comment]
Hi @pieterdd , preliminary question: are you 100% sure you are not loading VST2.x plug-ins? Giada only supports VST3.

**Reply by pieterdd** [reply]
Hi @gvnnz. Here's a snippet of the directory listing:

```
❯ ls -1 /Library/Audio/Plug-Ins/VST3/
...
drwxr-xr-x  3 root    wheel   96 Feb 17  2021 Vital.vst3
drwxr-xr-x@ 3 pieter  wheel   96 Feb 22  2021 Youlean Loudness Meter 2.vst3
```

Interestingly, they all seem to be directories. Anything I should check to ensure these are actual VST3 plugins and not just things with a .VST3 extension? In Reaper, these plugins show up for me under the VST3/VST3i categories.

**Comment by gvnnz** [comment]
VST3 plug-ins on macOS are distributed as packages, they look as regular directories if scanned with `ls` - so what you see is correct here. Could you please paste here the debug output from Giada when started from the command line?

**Reply by pieterdd** [reply]
Assuming `/Applications/giada.app/Contents/MacOS/giada` is the correct way to start Giada from the command line on macOS, here's the full log:

```
❯ /Applications/giada.app/Contents/MacOS/giada
[init] Giada 0.18.2
[init] Build date: Sep 12 2021
[init] Release build
[init] Dependencies:
[init]   FLTK - 1.3.7
[init]   RtAudio - 5.1.0
[init]   RtMidi - 3.0.0
[init]   Libsamplerate
[init]   Libsndfile - libsndfile-1.0.31
[init]   JSON for modern C++ - 3.9.1
[init]   JUCE - 6.1.0
[KA] Compiled RtAudio APIs: 1
  CoreAudio
[midiMapConf::init] scanning midimaps directory '/Users/pieter/Library/Application Support/Giada/midimaps/'...
[midiMapConf::init] unable to scan midimaps directory!
```

I started the application, went to settings, did a scan of the VST3 folder, then quit the program.

**Reply by gvnnz** [reply]
Thanks a lot @pieterdd , that's the correct way to start Giada. Could you please also paste here the cmd output during the plug-ins scan?

**Reply by pieterdd** [reply]
This is the full log including the scan :) The scan doesn't produce any new log output - or at least not when I tried it. This is the only form of feedback I see:

![Screen Shot 2021-10-04 at 10 28 38](https://user-images.githubusercontent.com/1407980/135818720-28230139-b46d-4c81-87b9-c75f9616750a.png)

At the end of the scan, it still says "Scan (0 found)" on the button.

**Reply by CarloCattano** [reply]
Hey, double check that you have under misc "debug messages to terminal" or something similar

**Reply by gvnnz** [reply]
Related issue: #512 . Investigation is underway.
